rir/src/ir/BC_inc.h

```diff
@@ -140,33 +140,16 @@ class BC {
 
     BC() : bc(Opcode::invalid_) {}
 
-    BC(Opcode* pc) {
-        bc = *pc;
-        pc++;
-        immediate = decodeImmediateArguments(bc, pc);
-        // Read implicit promise argument offsets
-        if (bc == Opcode::call_implicit_ ||
-            bc == Opcode::named_call_implicit_) {
-            pc += sizeof(CallFixedArgs);
-            for (size_t i = 0; i < immediate.callFixedArgs.nargs; ++i)
-                immediateCallArguments.push_back(readImmediate(&pc));
-        }
-        // Read named arguments
-        if (bc == Opcode::named_call_ || bc == Opcode::named_call_implicit_) {
-            for (size_t i = 0; i < immediate.callFixedArgs.nargs; ++i)
-                callArgumentNames.push_back(readImmediate(&pc));
-        }
-    }
-
-    BC operator=(BC other) {
-        bc = other.bc;
-        immediate = other.immediate;
-        return other;
-    }
+    BC& operator=(BC other) = delete;
+    BC(const BC& other) = delete;
+    BC(const BC&& other)
+        : bc(other.bc), immediate(other.immediate),
+          immediateCallArguments(std::move(other.immediateCallArguments)),
```

Comment:
`std::swap`?

Reply:
I don't think that's what we want here

Reply:
I was thinking just for the two vectors. Why wouldn't it work? (just curious, `move` is fine too)